group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1685,43 +1702,17 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, CoordinatorRecord>
             // The subscription metadata is updated in two cases:
             // 1) The member has updated its subscriptions;
             // 2) The refresh deadline has been reached.
-            Map<String, Integer> subscribedTopicNamesMap = group.computeSubscribedTopicNames(
+            UpdateSubscriptionMetadataResult result = updateSubscriptionMetadata(
```

Comment:
This is a small refactoring to share more code between the consumer and the classic groups.